### ISSUE TITLE
pubsub: Nack not redelivered ijn pstest test

### ISSUE THREAD

**Issue by atombender** [googleapis/google-cloud-go#10707]
## Client

PubSub client 1.39 and pstest.

## Environment

Go 1.22.

## Code and Dependencies

I've created a repro [here](https://github.com/atombender/pubsub-bug-repro). Run with:

```shell
$ go test -v . -count=10
```

You'll see frequent failures:

```
=== RUN   TestBug
    bug_test.go:70: received message
    bug_test.go:74: nacking
    bug_test.go:82:
        	Error Trace:	bug_test.go:82
        	Error:      	Not equal:
        	            	expected: 2
        	            	actual  : 1
        	Test:       	TestBug
```

## Additional context

Confirmed to only happen >= 1.39. The cause may be [this change](https://github.com/googleapis/google-cloud-go/pull/10234/files#diff-4b433806a676f9e427cace393434175aaa9c8642472b468e400607c62649c2c7R318-R325) to the non-exactly-once delivery internals.

**Comment by hongalex** [comment]
Yeah can confirm that the linked change was what resulted in the regression since the previous commit `21cba147834634892a8d8b264bd48e235bd50091` doesn't have this problem.

What that change does is batch the receipt modacks (the first modack the client library issues upon receipt of the message). Since there's a slight delay in the receipt modack because of batching, I suspect what is happening is that there is a race condition between whether it goes [modack -> nack] or [nack -> modack]. The first case was what was happening before, but in the 2nd case, the message isn't redelivered until the modack expires.

This only affects the fake because in the actual server, messages are identified by a unique `ackID` per delivery, whereas in the fake the `ackID` is just the messageID.

I need to think more about what is the proper fix here. Introducing unique ackIDs per delivery might cause too much trouble, so I'm trying to avoid that if possible.

**Comment by hongalex** [comment]
I'll be downgrading the priority since fixing the fake behavior to fully resolve the issue (race condition) isn't possible at the moment.

**Comment by feywind** [comment]
Is this still problematic? @hongalex Do we still want to try to fix this, or is the sleep a good enough solution?
